lib/lz4.c

```diff
@@ -249,6 +249,9 @@ static int g_debuglog_enable = 1;
   typedef uint64_t U64;
   typedef uintptr_t uptrval;
 #else
+#if defined(__SIZEOF_INT__) && (__SIZEOF_INT__ != 4)
```

Comment:
Yes, this static test is welcomed.

nit picking here, but I think that `__SIZEOF_INT__` is `gcc`, and not part of the `C` standard. There is no problem if this macro is not defined, your code protects this situation, and just disables this test.

Another way to check the same thing would be to probe that `UINT_MAX`, defined in `<limits.h>`, has value `4294967295`. It's equivalent, but `<limits.h>` is `C89`, hence more portable.

Reply:
Changed the check to look at UINT_MAX.